### PR TITLE
feat: take dynamic asset id into account with `forc-client`

### PR DESCRIPTION
**Pull request by kayagokalp**
## Description
Base asset ids are dynamic for the newer fuel-core versions. This PR adds capability to query that and use it for forc-client transactions. Also adds a flag `max-fee` which is required to be set for newer versions of fuel-core.

### CODE REVIEW

File: forc-plugins/forc-client/src/util/tx.rs [hunk 1]
```diff
@@ -311,6 +309,8 @@ impl<Tx: Buildable + field::Witnesses + Send> TransactionBuilderExt<Tx> for Tran
                 e
             })?;
 
+        // Set max_fee from consensus_parameters.
```
Comment (sdankel) [inline forc-plugins/forc-client/src/util/tx.rs:312]: Is that what the section below is doing? I don't see a reference to `max_fee` in lines 314-326

**Review by sdankel — COMMENTED** [review]
Could we add a unit test or two for this?

**Review by sdankel — DISMISSED** [review]

File: forc-plugins/forc-client/src/util/tx.rs (hunk 1 above)
Comment (kayagokalp) [inline forc-plugins/forc-client/src/util/tx.rs:312]:
```suggestion
```

Comment (kayagokalp) [reply forc-plugins/forc-client/src/util/tx.rs]: Opps nice catch thanks

**Comment by kayagokalp** [comment]
> Could we add a unit test or two for this?

Opened an issue to track but relaying here as well, unfortunately it is very complicated to write complicated tests to forc-client as it requires multiple different network instantiations, hopefully once we have some headroom we can create this infra and test going forward

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**